File: src/config/supabase.js
import { createClient } from '@supabase/supabase-js';

const url = process.env.SUPABASE_URL;
const anonKey = process.env.SUPABASE_ANON_KEY;
const serviceKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!url || !anonKey || !serviceKey) {
  throw new Error('Missing Supabase environment variables. Check your .env file.');
}

// Used for user-scoped queries (respects Row Level Security)
export const supabase = createClient(url, anonKey);

// Used for admin operations (bypasses RLS) — never expose this to the client
export const supabaseAdmin = createClient(url, serviceKey);

File: app/config/supabase.py
from supabase import create_client, Client
from app.config.settings import settings

# Anon client — respects Row Level Security
supabase: Client = create_client(settings.SUPABASE_URL, settings.SUPABASE_ANON_KEY)

# Admin client — bypasses RLS (never expose to frontend)
supabase_admin: Client = create_client(settings.SUPABASE_URL, settings.SUPABASE_SERVICE_ROLE_KEY)
